#ifndef GUI_NEW_DEAL_DLG_H
#define GUI_NEW_DEAL_DLG_H

////////////////////////////////////////////////////////////////////////////////
//
// Includes
//
#ifndef FM_CORE_H
#	include "fm_core.h"
#endif

#ifndef GUI_STORE_EXPLORER_WDG_H
#	include "gui_storeexplorer_wdg.h"
#endif

// Ui
#include "ui_newdeal.h"

// Qt Includes
#include <QDialog>
////////////////////////////////////////////////////////////////////////////////

////////////////////////////////////////////////////////////////////////////////
namespace fm {
////////////////////////////////////////////////////////////////////////////////
namespace gui {
////////////////////////////////////////////////////////////////////////////////

////////////////////////////////////////////////////////////////////////////////
//
//	Storage Cpdateing Configuration
//
struct SProductInfo
{
	QString sName;
	int     nCount;
};

typedef QList<SProductInfo>          CStoreageUpdateConfig;
typedef QList<CStoreageUpdateConfig> CFarmUpdateConfig;
////////////////////////////////////////////////////////////////////////////////


////////////////////////////////////////////////////////////////////////////////
//
// class CAddCustomerDlg
//
class CNewDealDlg : public QDialog
{
	Q_OBJECT

public:// Constructors
	CNewDealDlg( QWidget* pwParent = nullptr );
	~CNewDealDlg() = default;

public:
	//
	//	Main Interface
	//
	// Clear Dialog
	inline void Clear();

public slots:
	//
	//	Slots
	//
	void OnCustomerChanged( int nIndex );

protected:// Helper Methodes
	void SetLabelColor( QLabel* pLabel, bool bValid );

protected:
	//
	//	Overidden Methods
	//
	QSize sizeHint() const override;

private:
	//
	//	Content
	//
	Ui::NewDealUI ui;
	CStoreExplorer* m_pwStoreExplorer;
};
////////////////////////////////////////////////////////////////////////////////

////////////////////////////////////////////////////////////////////////////////
//
//	Inline Implementations
//
////////////////////////////////////////////////////////////////////////////////

// Clear
inline void CNewDealDlg::Clear()
{
	ui.editFirstName->clear();
	ui.editLastName->clear();
	ui.editPhone->clear();
}

////////////////////////////////////////////////////////////////////////////////
} // namespace gui
////////////////////////////////////////////////////////////////////////////////
} // namespace fm
////////////////////////////////////////////////////////////////////////////////
#endif // GUI_NEW_DEAL_DLG_H